corehq/apps/consumer_user/signals.py

```diff
@@ -0,0 +1,45 @@
+from corehq.apps.consumer_user.models import (
+    ConsumerUserCaseRelationship,
+    ConsumerUserInvitation,
+)
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.form_processor.signals import sql_case_post_save
+
+from .const import (
+    CONSUMER_INVITATION_ACCEPTED,
+    CONSUMER_INVITATION_CASE_TYPE,
+    CONSUMER_INVITATION_SENT,
+    CONSUMER_INVITATION_STATUS,
+)
+from .tasks import create_new_invitation
+
+
+def send_email_case_changed_receiver(sender, case, **kwargs):
+    if case.type != CONSUMER_INVITATION_CASE_TYPE:
+        return
+    email = case.get_case_property('email')
+    status = case.get_case_property(CONSUMER_INVITATION_STATUS)
+    invitation = ConsumerUserInvitation.objects.filter(case_id=case.case_id,
```

Comment:
I'm not sure how I didn't catch this before, but this is the wrong case_id. From the spec, this should be the "uuid for the patient/demographic case."
It would be something like:

```python
host_indices = [index for index in case.indices where index.relationship = 'extension']
try:
    person_case_id = host_indices[0].referenced_id
except IndexError:
    return    # the app is set up incorrectly.
```

Reply:
We only save case_id, domain in both ConsumerUserInvitation and ConsumerUserCaseRelationship models. You want me to store uuid for the patient/demographic case in the case_id right. I tried implementing the above logic but getting host_indices as empty list when I call from tests using create_case. I tried looking at other usages but could n't find anything helpful. Can you share here how to use this correctly?

Also, Since we are now planning to use uuid for the patient/demographic case in the case_id, will update_case work with this reference_id? Should I add a new column in both ConsumerUserInvitation and ConsumerUserCaseRelationship models to store case_id too?

Reply:
You can create indexed cases like this:

https://github.com/challabeehyv/commcare-hq/blob/consumer_user_poc/corehq/ex-submodules/casexml/apps/case/tests/test_db_accessors.py#L178-L186

So you would need to make both a demographic, and an invitation case in your tests. 

We want the CaseRelationship to link to the demographic case, and we would want to store that id on the invitation when it is created to stop a case getting reassigned between the invitation being created and being accepted.

I think if there can only be a single invitation case open for a particular demographic case, we should be able to always get the invitation back (by looking at the reverse indexed cases for one with the invitation type). 

However, I agree that it might be better to store both invitation case_id and the demographic case_id on the `ConsumerUserInvitation`. I don't think you need to add this to the `ConsumerUserCaseRelationship`.